Source/com/Communicator.h

```diff
@@ -437,6 +437,10 @@ namespace RPC {
     };
 
     class EXTERNAL Communicator {
+    private:
+        static constexpr uint8_t SoftKillCheckWaitTime = 10; // In seconds
```

Comment:
Think these are not used at the moment, right?
But perhaps it is a good alternative to passing these values around everywhere. We could make them static here (not constexpression) I guess and then set them (override basically then in that case) from the Communicator constructor. These values will never change, they are read from the config one and then never changed at runtime and they are the same for all usages, or am I missing something. And although they will never change anymore before even used, these are bytes even if they would change while being used that would be safe and their usage pattern (just single values) would not require any memory barrier. By having them static we do not have to copy these rather const values around to all classes/objects. They can just be used from a static method from the Communicator class everywhere else in this file.

Reply:
yes these are used as the default values to assign from the communicator constructor. 

Communicator::Communicator(const Core::NodeId& node, const string& proxyStubPath)
         : _connectionMap(*this)
         , _ipcServer(node, _connectionMap, proxyStubPath)
         _, _softKillCheckWaitTime(SoftKillCheckWaitTime)
         , _hardKillCheckWaitTime(HardKillCheckWaitTime)_